test/gpuarrayscore.jl

```diff
@@ -0,0 +1,9 @@
+using Test, GPUArraysCore, EnzymeCore
+
+@testset "EnzymeCore" begin
+    @test nothing == EnzymeCore.EnzymeRules.inactive_noinl(GPUArraysCore.assertscalar)
+
+    @test nothing == EnzymeCore.EnzymeRules.inactive_noinl(GPUArraysCore.default_scalar_indexing)
+
+    @test nothing == EnzymeCore.EnzymeRules.inactive_noinl(GPUArraysCore.allowscalar, identity)
```

Comment:
Testing for the exact definition of interface methods isn't good testing. We don't want to have to update tests every time implementation details (of Enzyme or GPUArrays) change. Instead, this should test for observed, meaningful behavior (i.e., why are these definitions needed in the first place).